flink-table/flink-sql-client/src/test/resources/sql/function.q

```diff
@@ -346,3 +346,159 @@ show user functions;
+|        class name |                                   $VAR_UDF_JAR_PATH_SPACE UpperUDF |
+| function language |                                       $VAR_UDF_JAR_PATH_SPACE JAVA |
+|     resource uris | [ResourceUri{resourceType=JAR, uri='$VAR_UDF_JAR_PATH'}] |
++-------------------+---------------------------------------------$VAR_UDF_JAR_PATH_DASH+
+5 rows in set
+!ok
+
+describe function extended temp_upperudf;
++-------------------+---------------------------------------------$VAR_UDF_JAR_PATH_DASH+
+|         info name |                                 $VAR_UDF_JAR_PATH_SPACE info value |
++-------------------+---------------------------------------------$VAR_UDF_JAR_PATH_DASH+
+|   system function |                                      $VAR_UDF_JAR_PATH_SPACE false |
+|         temporary |                                       $VAR_UDF_JAR_PATH_SPACE true |
+|        class name |                                   $VAR_UDF_JAR_PATH_SPACE UpperUDF |
+| function language |                                       $VAR_UDF_JAR_PATH_SPACE JAVA |
+|     resource uris | [ResourceUri{resourceType=JAR, uri='$VAR_UDF_JAR_PATH'}] |
+|              kind |                                     $VAR_UDF_JAR_PATH_SPACE SCALAR |
+|      requirements |                                         $VAR_UDF_JAR_PATH_SPACE [] |
+|     deterministic |                                       $VAR_UDF_JAR_PATH_SPACE true |
+|  constant folding |                                       $VAR_UDF_JAR_PATH_SPACE true |
+|         signature |                $VAR_UDF_JAR_PATH_SPACE c1.db.temp_upperudf(STRING) |
++-------------------+---------------------------------------------$VAR_UDF_JAR_PATH_DASH+
+10 rows in set
+!ok
+
+desc function temp_upperudf;
++-------------------+---------------------------------------------$VAR_UDF_JAR_PATH_DASH+
+|         info name |                                 $VAR_UDF_JAR_PATH_SPACE info value |
++-------------------+---------------------------------------------$VAR_UDF_JAR_PATH_DASH+
+|   system function |                                      $VAR_UDF_JAR_PATH_SPACE false |
+|         temporary |                                       $VAR_UDF_JAR_PATH_SPACE true |
+|        class name |                                   $VAR_UDF_JAR_PATH_SPACE UpperUDF |
+| function language |                                       $VAR_UDF_JAR_PATH_SPACE JAVA |
+|     resource uris | [ResourceUri{resourceType=JAR, uri='$VAR_UDF_JAR_PATH'}] |
++-------------------+---------------------------------------------$VAR_UDF_JAR_PATH_DASH+
+5 rows in set
+!ok
+
+desc function extended temp_upperudf;
++-------------------+---------------------------------------------$VAR_UDF_JAR_PATH_DASH+
+|         info name |                                 $VAR_UDF_JAR_PATH_SPACE info value |
++-------------------+---------------------------------------------$VAR_UDF_JAR_PATH_DASH+
+|   system function |                                      $VAR_UDF_JAR_PATH_SPACE false |
```

Comment:
make order determistic i.e. sort by first column

Reply:
is that necessary? we don't do that for `DESCRIBE CATALOG` for example. I tried to order the rows in a way that I thought was logical

Reply:
I agree with @nateab, if required let's change both for `DESCRBIE CATALOG` and `FUNCTION` in a follow up.